partner_identification_unique_by_category/tests/test_partner_identification_unique_by_category.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo.exceptions import ValidationError
+from odoo.tests.common import SavepointCase
+
+
+class TestPartnerIdentificationUniqueByCategory(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.category_1 = cls.env["res.partner.id_category"].create(
+            {"code": "UID", "name": "Group Id"}
+        )
+        cls.category_2 = cls.env["res.partner.id_category"].create(
+            {"code": "GID", "name": "User Id"}
+        )
+        cls.partner_1 = cls.env.ref("base.res_partner_1")
+        cls.partner_2 = cls.env.ref("base.res_partner_2")
+
+    def test_id_creation(self):
+        """Check Id unique (or not) validation."""
+        self.category_1.has_unique_numbers = True
+        self.env["res.partner.id_number"].create(
+            {
+                "name": "123",
+                "category_id": self.category_1.id,
+                "partner_id": self.partner_1.id,
+            }
+        )
+        with self.assertRaises(ValidationError):
```

Comment:
use `assertRaisesRegex` to make sure you get your own exception

Reply:
Done

Reply:
not for all exception tests :wink: